gtda/plotting/persistence_diagrams.py

```diff
@@ -26,61 +26,71 @@ def plot_diagram(diagram, homology_dimensions=None, **input_layout):
     if homology_dimensions is None:
         homology_dimensions = np.unique(diagram[:, 2])
 
-    max_filt_param = np.where(np.isinf(diagram), -np.inf, diagram).max()
+    diagram_no_dims = diagram[:, :2]
+    max_birth, max_death = np.where(
+        np.isposinf(diagram_no_dims), -np.inf, diagram_no_dims
+    ).max(axis=0)
+    min_birth, min_death = np.where(
+        np.isneginf(diagram_no_dims), np.inf, diagram_no_dims
+    ).min(axis=0)
 
-    layout = dict(
+    fig = gobj.Figure()
+    fig.add_trace(gobj.Scatter(x=[100 * min(-np.abs(max_death), min_birth),
+                                  100 * max_death],
+                               y=[100 * min(-np.abs(max_death), min_birth),
+                                  100 * max_death],
+                               mode='lines',
+                               line=dict(dash='dash', width=1, color='black'),
+                               showlegend=False, hoverinfo='none'))
+
+    for dim in homology_dimensions:
+        name = f'H{int(dim)}' if dim != np.inf else 'Any homology dimension'
+        subdiagram = _subdiagrams(np.asarray([diagram]), [dim],
+                                  remove_dim=True)[0]
+        diff = (subdiagram[:, 1] != subdiagram[:, 0])
```

Comment:
this is a comparison between floats, right? Have you considered using `np.isclose`?

Reply:
Are you commenting on line 50?  There, I am ignoring points on the diagonal.  In `giotto-tda`, those are guaranteed to be created as exactly the same floats.

Reply:
I was commenting on 50.
Is it to filter out the points that we might be padding the persistence diagrams with?

Reply:
Exactly!